File: 1343.大小为-k-且平均值大于等于阈值的子数组数目.go
/*
 * @lc app=leetcode.cn id=1343 lang=golang
 *
 * [1343] 大小为 K 且平均值大于等于阈值的子数组数目
 */

// @lc code=start
func numOfSubarrays(arr []int, k int, threshold int) int {
	res, t := 0, 0
	l, k := 0, k
	for i := 0; i < k; i++ {
		t += arr[i]
	}
	for r <= len(arr) {
		if t >= k*threshold {
			res++
		}
		t -= arr[l]
		l++
		if r < len(arr) {
			t += arr[r]
		}
		r++
	}
	return res

}

// @lc code=end

File: 1343.大小为-k-且平均值大于等于阈值的子数组数目.py
#
# @lc app=leetcode.cn id=1343 lang=python3
#
# [1343] 大小为 K 且平均值大于等于阈值的子数组数目
#

# @lc code=start
class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        res = 0
        l, r = 0, k
        t = sum(arr[:k])
        while r <= len(arr):
            if t >= k * threshold:
                res += 1
            t -= arr[l]
            l += 1
            if r < len(arr):
                t += arr[r]
            r += 1
            
        return res
# @lc code=end

